hudi-flink/src/main/java/org/apache/hudi/sink/transform/RowDataToHoodieFunction.java

```diff
@@ -89,11 +96,19 @@ public void open(Configuration parameters) throws Exception {
         HoodieAvroKeyGeneratorFactory
             .createKeyGenerator(flinkConf2TypedProperties(FlinkOptions.flatOptions(this.config)));
     this.payloadCreation = PayloadCreation.instance(config);
+    long totalLimit = this.config.getLong(FlinkOptions.WRITE_RATE_LIMIT);
+    if (totalLimit > 0) {
+      this.rateLimiter = RateLimiter.create((double) totalLimit / getRuntimeContext().getNumberOfParallelSubtasks());
+    }
   }
 
   @SuppressWarnings("unchecked")
   @Override
   public O map(I i) throws Exception {
+    // If rateLimiter is enabled, forward flow control is performed on the data.
+    if (rateLimiter != null) {
```

Comment:
can we use option instead of null?

Reply:
Or better, we can use a different `RowDataToHoodieFunction` impl when rate limi turns on.

Reply:
@garyli1019 @danny0405 Thanks for your comments and suggestions on this PR.